service.toml

```diff
@@ -20,30 +20,92 @@ implement = ["multiparter"]
 
 [namespace.storage.op.delete]
-optional = ["multipart_id"]
+optional = ["multipart_id", "excepted_bucket_owner"]
 
 [namespace.storage.op.list]
 optional = ["list_mode"]
 
 [namespace.storage.op.read]
-optional = ["offset", "io_callback", "size"]
+optional = ["offset", "io_callback", "size", "excepted_bucket_owner", "sse_customer_algorithm", "sse_customer_key", "sse_customer_key_md5"]
 
 [namespace.storage.op.write]
-optional = ["content_md5", "content_type", "io_callback", "storage_class"]
+optional = ["content_md5", "content_type", "io_callback", "storage_class", "excepted_bucket_owner", "bucket_key_enabled", "sse_customer_algorithm", "sse_customer_key", "sse_customer_key_md5", "sse_kms_key_id", "server_side_encryption"]
+
+[namespace.storage.op.stat]
+optional = ["excepted_bucket_owner", "sse_customer_algorithm", "sse_customer_key", "sse_customer_key_md5"]
+
+[namespace.storage.op.create_multipart]
+optional = ["bucket_key_enabled", "excepted_bucket_owner", "sse_customer_algorithm", "sse_customer_key", "sse_customer_key_md5", "sse_kms_key_id", "server_side_encryption"]
+
+[namespace.storage.op.write_multipart]
+optional = ["excepted_bucket_owner", "sse_customer_algorithm", "sse_customer_key", "sse_customer_key_md5"]
+
+[namespace.storage.op.complete_multipart]
+optional = ["excepted_bucket_owner"]
 
 [pairs.default_service_pairs]
 type = "DefaultServicePairs"
 description = "set default pairs for service actions"
 
+[pairs.force_path_style]
+type = "bool"
+description = "see http://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html for Amazon S3: Virtual Hosting of Buckets"
+
+[pairs.disable_100_continue]
+type = "bool"
+description = "set this to `true` to disable the SDK adding the `Expect: 100-Continue` header to PUT requests over 2MB of content"
+
+[pairs.use_accelerate]
+type = "bool"
+description = "set this to `true` to enable S3 Accelerate feature"
+
+[pairs.disable_content_md5_validation]
+type = "bool"
+description = "set this to `true` to disable the S3 service client from automatically adding the ContentMD5 to S3 Object Put and Upload API calls"
+
+[pairs.use_arn_region]
+type = "bool"
+description = "set this to `true` to have the S3 service client to use the region specified in the ARN, when an ARN is provided as an argument to a bucket parameter"
+
 [pairs.default_storage_pairs]
 type = "DefaultStoragePairs"
 description = "set default pairs for storager actions"
 
 [pairs.storage_class]
 type = "string"
 
+[pairs.bucket_key_enabled]
+type = "bool"
+
+[pairs.excepted_bucket_owner]
+type = "string"
+description = "the account ID of the excepted bucket owner"
+
+[pairs.sse_customer_algorithm]
+type = "string"
+description = "specifies the algorithm to use to when encrypting the object"
+
+[pairs.sse_customer_key]
+type = "string"
+description = "specifies the customer-provided encryption key for Amazon S3 to use to decrypt the source object"
+
+[pairs.sse_customer_key_md5]
```

Comment:
Let's remove the `sse_customer_key_md5` and calculate for users before sending requests.